### PR TITLE
Add wrapper for msk stack, to expose bootstap host list

### PR DESCRIPTION
**Pull request by reedstrm**
This adds a wrapper for MSK::Cluster resources, to lookup and return the bootstrap host list, which is what is needed for Kafka clients (like the quasar event-capture-api server) to connect to the msk cluster.

### CODE REVIEW

**Comment by reedstrm** [comment]
This still needs specs for the msk cluster wrapper

File: spec/msk_cluster_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'spec_helper'
+
+RSpec.describe OpenStax::Aws::MskCluster do
+  let(:instance) { described_class.new(msk_instance_identifier: 'foo', region: 'us-east-1') }
+
+  it 'Retrieves list of bootstrap hosts' do
+      expect(instance.client).to receive(:get_bootstrap_brokers).with(cluster_arn: 'foo')
+    instance.bootstrap_hosts
+  end
```
Comment (reedstrm) [inline spec/msk_cluster_spec.rb:9]: Do I need to mock this, or use a real cluster name and VCR record the interactions/fetching data?

File: lib/openstax/aws/msk_cluster.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    attr_reader :client, :msk_instance_identifier
+
+    delegate_missing_to :@client
+
+    def initialize(msk_instance_identifier:, region:)
+      @msk_instance_identifier = msk_instance_identifier
+      @client = ::Aws::Kafka::Client.new(region: region)
+    end
+
+    def bootstrap_hosts()
```
Comment (steveburkett) [inline lib/openstax/aws/msk_cluster.rb:13]: you dont need () here.  

https://github.com/airbnb/ruby#no-args-parens

File: lib/openstax/aws/stack.rb [hunk 3]
```diff
@@ -65,9 +65,14 @@ def template
       end
     end
 
-    def create(params: {}, wait: false)
+    def create(params: {}, wait: false, skip_if_exists: false)
       logger.info("**** DRY RUN ****") if dry_run
 
+      if skip_if_exists and exists?
```
Comment (steveburkett) [inline lib/openstax/aws/stack.rb:71]: why "and" vs "&&"?  the ruby convention is to use && for boolean expressions like this. Using and can often lead to order precedence issues that can be difficult to debut.

File: lib/openstax/aws/stack.rb [hunk 4]
```diff
@@ -65,9 +65,14 @@ def template
...
+        logger.info("Skipping #{name} stack - exists...")
+        return
```
Comment (steveburkett) [inline lib/openstax/aws/stack.rb:73]: perhaps it makes sense to return something here?  like false or an error object?  from the caller perspective, it seems confusing to know if the expectation of create succeeded or not.

File: lib/openstax/aws/stack.rb (hunk 3 above)
Comment (reedstrm) [reply lib/openstax/aws/stack.rb:71]: Yeah, same - python habits.

File: lib/openstax/aws/stack.rb (hunk 4 above)
Comment (reedstrm) [reply lib/openstax/aws/stack.rb:73]: the flag is basically for idempotent scripting: safely call to be sure it exists, either previously, or was created now. Note that this only happens when a dev explicitly uses an extra switch, so presumable will see the log, like this:
```
$ ./create_env  --env_name rjr-test --env_type dev --image_id ami-03945891ff05cc360 --do_it --try_harder
[2020-08-27 13:31:03.938] INFO  | Skipping rjr-test-quasar-network stack - exists...
[2020-08-27 13:31:04.020] INFO  | Skipping rjr-test-quasar-msk stack - exists...
[2020-08-27 13:31:04.110] INFO  | Skipping rjr-test-quasar-api stack - exists...
```

File: lib/openstax/aws/msk_cluster.rb (hunk 2 above)
Comment (reedstrm) [reply lib/openstax/aws/msk_cluster.rb:13]: python bleed through, probably. :)

File: lib/openstax/aws/stack.rb (hunk 4 above)
Comment (steveburkett) [reply lib/openstax/aws/stack.rb:73]: what about the return value?  here...the caller of create has no idea if create succeeded for failed and for any reason.  do you feel this isnt necessary?

File: spec/msk_cluster_spec.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+ACTUAL_MSK_ARN = 'arn:aws:kafka:us-east-2:373045849756:cluster/rjr-test-quasar/430c3363-00b0-4576-ae82-6894eb9ea401-3'
+RSpec.describe OpenStax::Aws::MskCluster, vcr: VCR_OPTS do
+  let(:instance) do
+    described_class.new(
+      cluster_arn: ACTUAL_MSK_ARN, region: 'us-east-2'
+    )
+  end
+
+  it 'Retrieves list of bootstrap hosts' do
```
Comment (steveburkett) [inline spec/msk_cluster_spec.rb:14]: minor comment.  it descriptions shouldn't be capitalized...it should read as a sentence.   https://www.betterspecs.org/

File: spec/msk_cluster_spec.rb (hunk 1 above)
Comment (reedstrm) [reply spec/msk_cluster_spec.rb]: fixed w/ vcr

File: lib/openstax/aws/msk_cluster.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
+module OpenStax::Aws
+  class MskCluster
+
+    attr_reader :client, :msk_instance_identifier
+
+    delegate_missing_to :@client
+
+    def initialize(msk_instance_identifier:, region:)
+      @msk_instance_identifier = msk_instance_identifier
```
[the remark this answers is not in the document]
Comment (reedstrm) [reply lib/openstax/aws/msk_cluster.rb]: fixed.

File: spec/msk_cluster_spec.rb (hunk 5 above)
Comment (reedstrm) [reply spec/msk_cluster_spec.rb:14]: thanks, will note for future

Comment (reedstrm) [reply spec/msk_cluster_spec.rb]: changed. (and improved. And moved the dang cassette, that uses the description as a filename. Oops)

**Pull request merged**